lib/puppet/type/grafana_datasource.rb

```diff
@@ -11,6 +11,10 @@
     desc 'The name of the datasource.'
   end
 
+  newparam(:uid) do
```

Comment:
I think this is actually a read-only property instead of a parameter??  As in I don't think you can specify it when creating a datasource.

Reply:
yes, you're right.  It _is_ possible to specify your own uuid.  I'll incorporate this into the refactor I'm working on.